src/core/time/duration.ts

```diff
@@ -0,0 +1,517 @@
+/**
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the ""License"");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an ""AS IS"" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import {IllegalArgumentError} from '../errors.js';
+import {MathEx} from '../util/math_ex.js';
+import {Time} from './time.js';
+
+/**
+ * A time-based amount of time, such as '34.5 seconds'.
+ *
+ * This class models a quantity or amount of time in terms of seconds and nanoseconds. It can be accessed using other
+ * duration-based units, such as minutes and hours. In addition, the DAYS unit can be used and is treated as exactly
+ * equal to 24 hours, thus ignoring daylight savings effects.
+ *
+ * A physical duration could be of infinite length. The duration uses nanosecond resolution with a maximum value of the
+ * seconds that can be held in a long. This is greater than the current estimated age of the universe.
+ *
+ * The range of a duration requires the storage of a number larger than a long. To achieve this, the class stores a long
+ * representing seconds and an integer representing nanosecond-of-second, which will always be between 0 and 999,999,999.
+ * The model is of a directed duration, meaning that the duration may be negative.
+ *
+ * The duration is measured in "seconds", but these are not necessarily identical to the scientific "SI second"
+ * definition based on atomic clocks. This difference only impacts durations measured near a leap-second and should not
+ * affect most applications.
+ *
+ * This is a value-based class; use of identity-sensitive operations on instances of Duration may have unpredictable
+ * results and should be avoided. The equals method should be used for comparisons.
+ */
+export class Duration {
+  /**
+   * A constant for a duration of zero.
+   */
+  public static readonly ZERO = new Duration(0, 0);
+
+  /**
+   * A constant for a duration of forever.
+   */
+  public static readonly FOREVER = new Duration(Number.MAX_SAFE_INTEGER, 999_999_999);
+
+  /**
+   * The number of seconds in this duration.
+   */
+  private readonly _seconds: number;
+
+  /**
+   * The number of nanoseconds in this duration.
+   */
+  private readonly _nanos: number;
+
+  /**
+   * Creates a new instance of Duration with the specified number of seconds and nanoseconds.
+   * This is a private constructor and not intended to be called directly.
+   *
+   * @param seconds - the number of seconds
+   * @param nanos - the number of nanoseconds
+   */
+  private constructor(seconds: number, nanos: number) {
```

Comment:
You can just define them as readonly and you will achieve the same effect without needing a getter :)

Reply:
Kotlin, Java, C/C++, and C# habits die hard.

Reply:
Fixed